0_meal_identification/meal_identification/meal_identification/datasets/dataset_operations.py

```diff
@@ -0,0 +1,211 @@
+    project_root = get_root_dir()
+    full_out_path_dir = os.path.join(project_root, output_dir)
+    os.makedirs(full_out_path_dir, exist_ok=True)  # Ensure the output directory exists
+
+    if include_gen_date_label:
+        filename = f"{data_gen_date}_{patient_id}_{data_label}.csv"
+    else:
+        filename = f"{patient_id}_{data_label}.csv"
+
+    file_path = os.path.join(full_out_path_dir, filename)
+    data.to_csv(file_path, index=True)
+    print(f"Data saved successfully in: {output_dir}")
+    print(f"\n \t Dataset label: {filename}")
+
+def dataset_label_modifier_fn(base_label_modifier,
+                              coerce_time, coerce_time_label,
+                              day_start_index_change, day_start_time_label,
+                              erase_meal_overlap, erase_meal_label):
+    """
+    Modify the data label based on applied transformations.
+
+    Parameters
+    ----------
+    base_label_modifier : str
+        The base label modifier for the dataset
+    coerce_time : bool
+        Whether to coerce the time interval of the data
+    coerce_time_label : str
+        The label modifier for the time interval coercion
+    day_start_index_change : bool
+        Whether to create a day index starting at a specific time
+    day_start_time_label : str
+        The label modifier for the day start time
+    erase_meal_overlap : bool
+        Whether to erase overlapping meals
+    erase_meal_label : str
+        The label modifier for the meal overlap erasure
+
+    Returns
+    -------
+    str
+        The modified data label
+    """
+    data_label_modifier = base_label_modifier
+    if coerce_time:
+        data_label_modifier += coerce_time_label
+    if day_start_index_change:
+        data_label_modifier += day_start_time_label
+    if erase_meal_overlap:
+        data_label_modifier += erase_meal_label
+
+    return data_label_modifier
+
+def coerce_time_fn(data, coerse_time_interval):
+    '''
+    Coerce the time interval of the data.
+
+    Parameters
+    ----------
+    data : pd.DataFrame
+        The input DataFrame with a 'date' column.
+    coerse_time_interval : pd.Timedelta
+        The interval for coarse time resampling.
+
+    Returns
+    -------
+    pd.DataFrame
+        The coerced DataFrame with a DatetimeIndex.
+    '''
+    # Ensure 'date' column exists
+    if 'date' not in data.columns:
+        raise KeyError("'date' column not found in data.")
+
+    # Convert 'date' to datetime if not already
+    if not pd.api.types.is_datetime64_any_dtype(data['date']):
+        data['date'] = pd.to_datetime(data['date'], errors='coerce')
+        data = data.dropna(subset=['date'])  # Drop rows where 'date' couldn't be parsed
+
+    # Set 'date' as the index without squeezing
+    data = data.set_index('date').sort_index()
+
+    # Define resample rule based on the provided timedelta
+    resample_rule = f'{int(coerse_time_interval.total_seconds() // 60)}min'  # e.g., '5min' for 5 minutes
```

Comment:
Get rid of this and follow https://pandas.pydata.org/docs/reference/api/pandas.Timedelta.html
for options. Let pandas handle the timing.